appserver/common/annotation-framework/src/main/java/org/glassfish/apf/impl/ComponentDefinition.java

```diff
@@ -41,12 +43,12 @@
  */
 public class ComponentDefinition implements ComponentInfo {
 
-    private static final Set<String> EXCLUDED_FROM_ANNOTATION_PROCESSING = Set.of(
+    private static final Set<String> EXCLUDED_FROM_ANNOTATION_PROCESSING = Collections.unmodifiableSet(new HashSet<>(Set.of(
```

Comment:
Set.of already returns unmodifialble collection, see javadoc ;)

Reply:
Aha, it is to allow contains(null), but ...

Reply:
[JDK-8265905: ImmutableCollections: Querying for 'null' should return 'false' instead of NPE](https://bugs.openjdk.org/browse/JDK-8265905) was closed as won't fix, so the only hope is for [JDK-8267823: Document NPE on immutable collections query methods](https://bugs.openjdk.org/browse/JDK-8267823). But it might take 10 LTSes to get there.